core/src/main/java/org/apache/accumulo/core/fate/zookeeper/FateLock.java

```diff
@@ -68,16 +72,34 @@ public FateLock(ZooReaderWriter zrw, FateLockPath path) {
     this.path = requireNonNull(path);
   }
 
+  public static class FateLockNode {
+    public final long sequence;
+    public final String lockData;
+
+    private FateLockNode(String nodeName) {
+      int len = nodeName.length();
+      Preconditions.checkArgument(nodeName.startsWith(PREFIX) && nodeName.charAt(len - 11) == '#',
+          "Illegal node name %s", nodeName);
+      sequence = Long.parseLong(nodeName.substring(len - 10));
+      lockData = nodeName.substring(PREFIX.length(), len - 11);
+    }
+  }
+
+  // TODO change data arg from byte[] to String.. in the rest of the code its always a String.
```

Comment:
I will open a follow on issue for this and remove the TODO before merging.

Reply:
Removed the TODO and #5264.  In the TODO I was thinking of using String, but realized using more concrete types would be better.